algorithms/polish.sbl

```diff
@@ -0,0 +1,240 @@
+/* Based on the Polish stemming algorithm developed by Błażej Kubiński,
+ * which can be found at https://github.com/Tutanchamon/pl_stemmer.
+ *
+ * The first Snowball implementation was written by Tomasz Łysiak, rewritten
+ * and modified by Dmitry Shachnev.
+ */
+
+stringescapes {}
+
+stringdef ak '{U+0105}' // ą a + ogonek
+stringdef ek '{U+0119}' // ę e + ogonek
+stringdef l/ '{U+0142}' // ł l + stroke
+stringdef s' '{U+015b}' // ś s + acute (kreska)
+stringdef c' '{U+0107}' // ć c + acute (kreska)
+stringdef o' '{U+00f3}' // ó o + acute (kreska)
+
+
+externals (stem)
+
+
+routines (
+  mark_regions
+  remove_nouns
+  remove_diminutive
+  remove_superlative_ends
+  remove_adjective_ends
+  remove_verbs_ends
+  remove_adverbs_ends
+  remove_plural_forms
+  remove_general_ends
+)
+
+integers ( pV p1 )
+
+groupings ( v )
+
+define v 'a{ak}e{ek}io{o'}uy'
+
+
+define mark_regions as (
+  $pV = limit
+  $p1 = limit
+
+  do (
+    gopast v     setmark pV
+    gopast non-v setmark p1
+  )
+)
+
+backwardmode (
+  define remove_nouns as (
+    setlimit tomark p1 for ([substring]) among (
+      'ach'  // plural locative
+      'ami'  // plural instrumental
+      'nia'  // genitive of words ending with -nie, e.g. czytanie - czytania
+      'niu'  // dative/locative of words ending with -nie, e.g. czytanie - czytaniu
+      'cia'  // genitive of words ending with -cie, e.g. ukrycie - ukrycia
+      'ciu'  // dative/locative of words ending with -cie, e.g. ukrycie - ukryciu
+      'ce'   // dative/locative of words ending with -ka, e.g. muzyka - muzyce
```

Comment:
I notice _muzyka_ stems to _muzyk_ while _muzyce_ stems to _muz_ - not necessary a problem, but if it's unexpected or more widespread this might be worth looking at.